tests/XeroOauth/Request/RequestHandlerTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+namespace DarrynTen\XeroOauth\Tests\XeroOauth\Request;
+
+use DarrynTen\XeroOauth\Request\RequestHandler;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use ReflectionClass;
+
+
+class RequestHandlerTest extends \PHPUnit_Framework_TestCase
+{
+    const TEST_URI = '/test-uri';
+
+    private $config = [
+        'key' => 'testKey',
+        'endpoint' => 'http://localhost:8082'
+    ];
+
+    /**
+     * @var RequestHandler
+     */
+    private $handler;
+
+    public function setUp()
+    {
+        $this->handler = new RequestHandler($this->config);
+    }
+
+    public function testInstanceOf()
+    {
+        $this->assertInstanceOf(RequestHandler::class, $this->handler, 'Handler must be an instance of '.RequestHandler::class);
+    }
+
+    /**
+     * @expectedException \DarrynTen\XeroOauth\Exception\ApiException
+     */
+    public function testWrongMethodHandleRequest()
+    {
+        $this->handler->handleRequest('WRONG', '/testUri', []);
+    }
+
+    /**
+     * @expectedException \DarrynTen\XeroOauth\Exception\ApiException
+     */
+    public function testGetHandlerRequestWithException()
+    {
+        $this->setUpMockClient(
+            new RequestException('Wrong request', new Request('GET', 'Bad one'), new Response(500))
+        );
+        $this->handler->handleRequest('GET', static::TEST_URI, []);
+    }
+
+    /**
+     * @dataProvider dataProvider
+     */
+    public function testHandleRequest($method, $uri, $result)
+    {
+        $this->setUpMockClient(
+            new Response(200, ['ContentType: application/json'], $result)
+        );
+
+        $this->assertEquals(
+            \GuzzleHttp\json_decode($result),
```

Comment:
It uses native php json_decode, but checks errors after a call and wraps them into an Exception. So it becomes more comfortable to use it.